docs/DeveloperGuide.md

```diff
@@ -137,55 +137,131 @@ Classes used by multiple components are in the `seedu.clinic.commons` package.
+<br>
+In case 2, `TYPE` needs to be set to `pw` or `ps`, and a `PRODUCT_NAME` needs to be specified.
+<br>
+<br>
+The deletion is limited to the items shown in the list displayed in GUI, and is done one item at a time.
+
+#### Path Execution of Delete Command
+
+The workflow of an `Delete` Command when it is executed by a user is shown in the activity diagram below:
+
+![Delete Command Activity Diagram](images/DeleteCommandActivityDiagram.png)
+<center><i>Figure n. Delete Command Activity Diagram</i></center>
+<br>
+
+With reference to the activity diagram above, the user input will be sent to **`DeleteCommandParser`** for parsing.
+The **`DeleteCommandParser`** will check if the compulsory prefixes are present (i.e. `ct/TYPE` and `i/INDEX`) and the corresponding
+argument values are all valid.
+A **`ParseException`** will be thrown if the check fails.
+
+If the `TYPE` parsed indicates a product deletion (via `ct/ps` or `ct/pw`), an additional field `PRODUCT_NAME` is parsed with its prefix and value checked.
+
+Next, a new **`DeleteCommand`** will be generated and executed. There are four possible paths for the `delete` command:
+
+1. The target to delete is a `supplier`/`warehouse` <br>
+CLI-nic finds the target `supplier`/`warehouse` at the specified `INDEX` of the displayed list, and remove it completely.
+
+1. The `INDEX` specified is invalid (e.g. exceeds the length of the list) <br>
+A **`CommandException`** error message wil be thrown.
+
+1. The target to delete is a `product` in a particular 'supplier' <br>
+CLI-nic finds the target `supplier` at the specified `INDEX` of the displayed supplier list, and retrieve its product list.<br>
+It finds the `product` with specified `PRODUCT_NAME` and remove it from the product list.
+
+1. No `product` has the `PRODUCT_NAME` in the target warehouse/supplier <br>
+A **`CommandException`** error message wil be thrown.
+
+#### Structure of Delete command
+
+We demonstrate the structure of the `delete` feature implementation below.
+
+![Delete Command Class Diagram](images/DeleteCommandClassDiagram.png)
+<center><i>Figure n. Delete Command Class Diagram</i></center>
+<br>
+
+Note that some commonly applied methods (like getter/setter methods for each attribute) are not included to reduce verbosity.
+
+#### Delete Command's interaction with related objects
+
+In the following section, the interaction between Delete Command and its associated objects in the delete feature will be discussed.
+The sequence diagrams below demonstrate the workflow in the deletion feature.
+
+![Delete Command Sequence Diagram](images/DeleteCommandSequenceDiagram.png)
+<center><i>Figure n. Delete Command Sequence Diagram for supplier deletion </i></center>
+<br>
+
+1. Parsing <br>
+
+    CLI-nic's `ClinicParser` will parse the user input and if the `delete` command word is present, the parser will try to parse the
+    input into a valid `DeleteCommand` via `DeleteCommandParser`. <br>
```

Comment:
Would "parse the input to generate a valid `DeleteCommand`" be better rather than "parse the input into a valid `DeleteCommand`"?

Reply:
That's right! Thanks for the suggestion!